TCAT/Controllers/HomeOptionsCardViewController.swift

```diff
@@ -0,0 +1,561 @@
+        }
+    }
+    
+    func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
+        if searchBar.showsCancelButton {
+            searchBarCancelButtonClicked(searchBar)
+        }
+    }
+}
+
+// MARK: HeaderView Delegate
+extension HomeOptionsCardViewController: HeaderViewDelegate {
+    func displayFavoritesTVC() {
+        if favorites.count < 2 {
+            presentFavoritesTVC()
+        } else {
+            let title = Constants.Alerts.MaxFavorites.title
+            let message = Constants.Alerts.MaxFavorites.message
+            let alertController = UIAlertController(title: title, message: message, preferredStyle: .alert)
+            let done = UIAlertAction(title: Constants.Alerts.MaxFavorites.action, style: .default)
+            alertController.addAction(done)
+            present(alertController, animated: true, completion: nil)
+        }
+    }
+    
+    func clearRecentSearches() {
+        SearchTableViewManager.shared.deleteAllRecents()
+        recentLocations = []
+        sections = createSections()
+    }
+}
+
+// MARK: MapView Delegate
+extension HomeOptionsCardViewController: HomeMapViewDelegate {
+    func reachabilityChanged(connection: Reachability.Connection) {
+        switch connection {
+        case .none:
+            isNetworkDown = true
+            searchBar.isUserInteractionEnabled = false
+            sections = []
+        case .cellular, .wifi:
+            isNetworkDown = false
+            sections = createSections()
+            searchBar.isUserInteractionEnabled = true
+        }
+    }
+}
+
+// MARK: TableView DataSource
+extension HomeOptionsCardViewController: UITableViewDataSource {
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return sections.count
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch sections[section].type {
+        case .seeAllStops: return 1
+        case .recentSearches: return recentLocations.count
+        case .favorites: return favorites.isEmpty ? 1 : favorites.count
+        case .searchResults: return sections[section].items.count
+        default: return 0
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        var cell: UITableViewCell!
+        
+        if sections[indexPath.section].type == .seeAllStops {
+            cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.seeAllStopsIdentifier) as? GeneralTableViewCell
+        }
+        // Favorites (including Add First Favorite!), Recent Searches
+        else {
+            guard let placeCell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.placeIdentifier) as? PlaceTableViewCell
+                else { return cell }
+            placeCell.place = sections[indexPath.section].items[indexPath.row]
+            cell = placeCell
+        }
+        
+        cell.layoutSubviews()
```

Comment:
why are we using `layoutSubviews`? should be doing constraints in the way that i mentioned in the other comment